hail/python/hail/table.py

```diff
@@ -1317,8 +1338,43 @@ def format_line(values, widths, right_align):
 
         return s
 
-    @typecheck_method(n=int, width=int, truncate=nullable(int), types=bool, handler=anyfunc)
-    def show(self, n=10, width=90, truncate=None, types=True, handler=print):
+    def _html_str(self, n, types):
+        import cgi
+
+        def hl_format(v):
+            return cgi.escape(Table._hl_repr(v))
+
+        t = self
+        t = t.flatten()
+        fields = list(t.row)
+
+        formatted_t = t.select(**{k: hl_format(v) for (k, v) in t.row.items()})
+        rows = formatted_t.take(n + 1)
+
+        has_more = len(rows) > n
+        rows = rows[:n]
+
+        def format_line(values):
+            return '<tr><td>' + '</td><td>'.join(values) + '</td></tr>\n'
+
+        s = '<table>'
+        s += '<thead style="font-weight: bold;">'
+        s += format_line(fields)
+        if types:
+            s += format_line([cgi.escape(str(t.row[f].dtype)) for f in fields])
+        s += '</thead><tbody>'
+        for row in rows:
+            s += format_line([row[f] for f in row])
+        s += '</tbody></table>'
+
+        if has_more:
+            n_rows = len(rows)
+            s += f"<p>showing top { n_rows } { 'row' if n_rows == 1 else 'rows' }</p>\n"
```

Comment:
we have a `plural` utility function in hail.utils.misc

Reply:
fixed